src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -347,6 +347,47 @@ def get_enable_msi_auth_for_monitoring(self) -> Union[bool, None]:
             return True
         return enable_msi_auth_for_monitoring
 
+    def _get_load_balancer_sku(self, enable_validation: bool = False) -> Union[str, None]:
+        """Internal function to obtain the value of load_balancer_sku, default value is
+        CONST_LOAD_BALANCER_SKU_STANDARD.
+
+        Note: When returning a string, it will always be lowercase.
+
+        This function supports the option of enable_validation. When enabled, it will check if load_balancer_sku equals
+        to CONST_LOAD_BALANCER_SKU_BASIC, if so, when api_server_authorized_ip_ranges is assigned or
+        enable_private_cluster is specified, raise an InvalidArgumentValueError.
+
+        :return: string or None
+        """
+        # read the original value passed by the command
+        load_balancer_sku = safe_lower(self.raw_param.get("load_balancer_sku"))
+        # try to read the property value corresponding to the parameter from the `mc` object
+        if (
+            load_balancer_sku is None and
+            self.mc and
+            self.mc.network_profile and
+            self.mc.network_profile.load_balancer_sku is not None
+        ):
+            load_balancer_sku = safe_lower(
+                self.mc.network_profile.load_balancer_sku
+            )
+        if self.decorator_mode == DecoratorMode.CREATE and load_balancer_sku is None:
+            load_balancer_sku = CONST_LOAD_BALANCER_SKU_STANDARD
```

Comment:
May consider adding this before the if statement in line 365, so that data is read from the mc model only when creating new cluster.

Reply:
we need to fetch original config from existing model when the cluster id updated.